lib/CommandCall.js

```diff
@@ -53,32 +44,26 @@ class CommandCall {
    * @returns {string}
    */
   toXML() {
-    if (this.type === 'sh') {
-      if (this.options) {
-        this.xml = iXmlNodeShOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error)
-              + this.command + iXmlNodeShClose();
-      } else {
-        this.xml = iXmlNodeShOpen('', '', '', '', '') + this.command + iXmlNodeShClose();
-      }
-    } else if (this.type === 'cl') {
-      const exec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
+    if (!availableCommands.includes(this.type)) {
+      throw new Error(`Invalid command type (${this.type}). Valid types: (${availableCommands.join(', ')})`);
+    }
 
-      if (this.options) {
-        this.xml = iXmlNodeCmdOpen(this.options.exec, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeCmdClose();
-      } else {
-        this.xml = iXmlNodeCmdOpen(exec, '', '', '', '', '') + this.command + iXmlNodeCmdClose();
-      }
-    } else if (this.type === 'qsh') {
-      if (this.options) {
-        this.xml = iXmlNodeQshOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeQshClose();
-      } else {
-        this.xml = iXmlNodeQshOpen('', '', '', '', '') + this.command + iXmlNodeQshClose();
-      }
+    if (this.type === 'cl') {
+      const defaultExec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
+      this.xml = `<cmd exec='${this.options.exec || defaultExec}'`;
+    } else {
+      this.xml = `<${this.type}`;
     }
 
+    // append optional values if set
+    if (this.options.rows) this.xml += ` rows='${this.options.rows}'`;
+    if (this.options.hex) this.xml += ` hex='${this.options.hex}'`;
+    if (this.options.before) this.xml += ` before='${this.options.before}'`;
+    if (this.options.after) this.xml += ` after='${this.options.after}'`;
+
+    const type = this.type === 'cl' ? 'cmd' : this.type;
```

Comment:
Oh, right. I forgot we needed the type for the closing tag. I would suggest setting the local variable above then:

```js
const type = this.type === 'cl' ? 'cmd' : this.type;
if(type == 'cmd') {
...
}
else {
...
}
```

Reply:
Could even do:
```js
const type = this.type === 'cl' ? 'cmd' : this.type;
this.xml = `<${this.type}`;

if (type == 'cmd') {
     const defaultExec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
     this.xml += ` exec='${this.options.exec || defaultExec}'`;
}
```